cmd/mtaBuild.go

```diff
@@ -261,65 +254,95 @@ func runMtaBuild(config mtaBuildOptions,
 	commonPipelineEnvironment.custom.mtaBuildToolDesc = filepath.ToSlash(mtaYamlFile)
 
 	if config.InstallArtifacts {
-		// install maven artifacts in local maven repo because `mbt build` executes `mvn package -B`
-		err = installMavenArtifacts(utils, config)
-		if err != nil {
+		if err = installMavenArtifacts(utils, config); err != nil {
 			return err
 		}
-		// mta-builder executes 'npm install --production', therefore we need 'npm ci/install' to install the dev-dependencies
-		err = utils.InstallAllDependencies(config.DefaultNpmRegistry)
-		if err != nil {
+		if err = utils.InstallAllDependencies(config.DefaultNpmRegistry); err != nil {
 			return err
 		}
 	}
 
 	if config.Publish {
-		log.Entry().Infof("publish detected")
-		if (len(config.MtaDeploymentRepositoryPassword) > 0) && (len(config.MtaDeploymentRepositoryUser) > 0) &&
-			(len(config.MtaDeploymentRepositoryURL) > 0) {
-			if (len(config.MtarGroup) > 0) && (len(config.Version) > 0) {
-				httpClient := &piperhttp.Client{}
-
-				credentialsEncoded := "Basic " + base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", config.MtaDeploymentRepositoryUser, config.MtaDeploymentRepositoryPassword)))
-				headers := http.Header{}
-				headers.Add("Authorization", credentialsEncoded)
+		if err = handlePublish(config, commonPipelineEnvironment, utils, mtarName, isMtarNativelySuffixed); err != nil {
```

Comment:
Maybe we can keep this log

log.Entry().Infof("publish detected")

Reply:
[b988bbd](https://github.com/SAP/jenkins-library/pull/5166/commits/b988bbd082505a8b7917291d1d7c03511ed26157)